### ISSUE TITLE
Questions about robustness

### ISSUE THREAD

**Issue by TanTingyi** [alexandrebarachant/pyRiemann#84]
Thank you very much for such an excellent job !
You have pointed out in your article the advantages of robustness based on Riemannian geometry and gave a sample example about MI.
But when I tried to apply this method to all the data of the datasets, I found that the total accuracy rate was  only close to 63%, there are many subjects' acc are 50%.
Can you give some suggestion what can I do now ?

**Comment by TanTingyi** [comment]
Here is my code:
```python
    
    train_X, train_y = data['session1'] # train data is session1
    test_X, test_y = data['session2'] # test data is session2

    covar= Covariances()
    tan = TangentSpace()
    clf = LogisticRegression()

    cov = covar.transform(train_X)
    cov_test = covar.transform(test_X)
    vec = tan.fit_transform(cov)  
    vec_test = tan.fit_transform(cov_test) 

    clf.fit(vec, train_y)
    predict = clf.predict(vec_test)
    acc = accuracy_score(test_y, predict)
    print(acc)
```

**Comment by sylvchev** [comment]
Thank you!
Do you know about [MOABB](https://github.com/NeuroTechX/moabb)? This project is made exactly to assess the robustness and the generalization power of different algorithms in BCI. There are examples of Riemannian approaches for MI, P300 and SSVEP, as well as other classical algorithm (CSP, XDAWN, CCA, ...).

Your code is correct, you could use sklearn [Pipelines](https://scikit-learn.org/stable/modules/compose.html) and [cross_val_score](https://scikit-learn.org/stable/modules/cross_validation.html) to automate the evaluation.

One problem that could lead to low accuracy is the covariance estimator, especially if there is not enough data. What is the shape of your `train_X` variable? You could use shrinkage estimator to obtain more robust results, try:
`covar= Covariances(estimator='lwf')  # 'oas'`
to use regularized estimators.

**Comment by mice-exp** [comment]
Hello, thank you so much for your dedication to the development of this tool!
I've been working with it in my research project, and I based my code on the MI example in this repository, I'm using a method combining  Riemannian Tangent Space + SVM, for ten subjects and the average is roughly 60% for MI (left/right hand). 
I'm not sure how else could I get better results. Is there a chance you could give me some suggestions? I would really appreciate it!!!


`runs =[] `
event_id = {}
data = read_raw_gdf(r"Desktop\data\SL\32SL_CSP.gdf",preload=True)
run_array = data['data']

for run in run_array:
    rawi = RawArray(data.get_data(),info=data.info, verbose=False)    
    runs.append(rawi)
raw = mne.concatenate_raws(runs)
raw.filter(l_freq=8, h_freq=30, method='iir')
events_from_annot, event_dict = mne.events_from_annotations(data)
event_id = dict(left=11, right=12)

epochs = mne.Epochs(raw, events_from_annot, event_id,tmin=0,tmax=2, baseline=None, picks=mne.pick_types(raw.info, eeg=True, stim=False))

X = epochs.get_data()
y = epochs.events[:,-1]; le = LabelEncoder(); y = le.fit_transform(y)

svc = SVC(kernel='rbf',decision_function_shape='ovr')

clf = make_pipeline(Covariances(estimator='oas'),TangentSpace(metric='riemann'),svc)


accuracy = cross_val_score(clf, X, y, cv=KFold(n_splits=5),scoring='roc_auc')`

Thank you for your time and help! 
Isabel

**Comment by DingkunLiu** [comment]
In most cases, XdawnCovariances works better for me.
